src/jit/lower.cpp

```diff
@@ -381,8 +383,47 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
-        break;
+        {
+            GenTree* parent = data->parent;
+            if (parent != nullptr)
+            {
+                // We do not need to force var = call() if the call is a top-level node.
```

Comment:
Please add this comment above 388.  The fact that we are forcing IR to be var=call only if GT_CALL node has a parent indicates that GT_CALL node is not a top-level stmnt.